### DISCUSSION TITLE
Download on Linux

### DISCUSSION THREAD

**Discussion by liviafabiano** [PlusToolkit/PlusLib#1218] [Q&A]
Hello everyone,
Which link of the listed PlusApp (Plus Server Launcher) should I download in order to use it in Linux system? As the pyigtl library is not supported for earlier version of Python than 3.6 and as Python 3.6 is not supported on Ubuntu 16.04, is there a Launcher I can download for (for instance) Ubuntu 20.04?
Thanks in advance

**Comment by adamrankin** [comment]
At this time there are only builds produced for Windows. For Linux you have to compile yourself.

**Reply by liviafabiano** [reply]
may I use these instruction? https://github.com/PlusToolkit/PlusBuild/blob/master/Docs/BuildInstructionsLinux.md

**Reply by lassoan** [reply]
> may I use these instruction? https://github.com/PlusToolkit/PlusBuild/blob/master/Docs/BuildInstructionsLinux.md

Yes.

Note that we plan to update the continuous integration infrastructure and as part of this we may start to build linux packages (in a few months).

**Reply by liviafabiano** [reply]
Perfect, thank you so much

**Reply by liviafabiano** [reply]
After properly compiling the PlusToolkit, I cannot find the Launcher in the bin folder. I just find 'PlusServer' and other files but not the launcher; how may I solve?

**Reply by adamrankin** [reply]
Did cmake find qt? Are the gui options enabled in cmake?

**Reply by liviafabiano** [reply]
I tried again to make the building but now it prints this error:
[ 95%] Linking CXX shared module ../../../lib/qml/VTK.9.1/libqmlvtkpluginD.so
Generating qmltypes file using qmlplugindump
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
QSocketNotifier: Can only be used with threads started with QThread
QQmlComponent: Component is not ready
make[5]: *** [GUISupport/QtQuick/qml/CMakeFiles/qmlvtkplugin.dir/build.make:150: lib/qml/VTK.9.1/libqmlvtkpluginD.so] Error 3
make[5]: *** Eliminazione del file «lib/qml/VTK.9.1/libqmlvtkpluginD.so»
make[4]: *** [CMakeFiles/Makefile2:11476: GUISupport/QtQuick/qml/CMakeFiles/qmlvtkplugin.dir/all] Error 2
make[3]: *** [Makefile:136: all] Error 2
make[2]: *** [CMakeFiles/vtk.dir/build.make:92: vtk-prefix/src/vtk-stamp/vtk-build] Error 2
make[1]: *** [CMakeFiles/Makefile2:125: CMakeFiles/vtk.dir/all] Error 2
make: *** [Makefile:101: all] Error 2

Is there maybe a conflict between qmlvtk versions?

**Reply by lassoan** [reply]
Could you please copy here the command that you use to build PLUS?

**Reply by liviafabiano** [reply]
As in the instruction for Ubuntu 22.04 I created the folder PlusBuild-bin and then
cmake ../PlusBuild
make

**Reply by lassoan** [reply]
OK, that looks good. The issue seems to be with QtQuick support in VTK. Maybe your Qt does not have QtQuick? You could either add QtQuick to your Qt installation, or disable QtQuick component in VTK.
